src/shogun/mathematics/Statistics.cpp

```diff
@@ -814,6 +814,171 @@ SGMatrix<float64_t> CStatistics::sample_from_gaussian(SGVector<float64_t> mean,
 	return S;
 }
 
+CStatistics::SigmoidParamters CStatistics::fit_sigmoid(SGVector<float64_t> scores, SGVector<float64_t> labels)
+{
+	SG_SDEBUG("entering CStatistics::fit_sigmoid()\n")
+
+	REQUIRE(scores.vector, "CStatistics::fit_sigmoid() requires "
+			"scores vector!\n");
+
+	/* count prior0 and prior1 if needed */
+	int32_t prior0=0;
+	int32_t prior1=0;
+	SG_SDEBUG("counting number of positive and negative labels\n")
+	{
+		for (index_t i=0; i<scores.vlen; ++i)
+		{
+			if (labels[i]>0)
+				prior1++;
+			else
+				prior0++;
+		}
+	}
+	SG_SDEBUG("%d pos; %d neg\n", prior1, prior0)
+
+	/* parameter setting */
+	/* maximum number of iterations */
+	index_t maxiter=100;
```

Comment:
all these parameters should be set somewhere else... as you want to be able to control them without actually changing the code... i.e. there should be a getter/setter for them

Reply:
I am not the one who originally implemented, I merely made a few corrections in `fit_sigmoid`. However, the code can certainly be improved. I'm on it.